spec/grape/endpoint_spec.rb

```diff
@@ -257,10 +257,29 @@ def app
         optional :third, default: 'third-default'
         optional :nested, type: Hash do
           optional :fourth
+          optional :fifth
+          optional :nested_two, type: Hash do
+            optional :sixth
+            optional :nested_three, type: Hash do
+              optional :seventh
+            end
+          end
         end
       end
     end
 
+    it 'should show nil for nested params if include_missing and include_missing_nested are true' do
+      inner_params = nil
+      subject.get '/declared' do
+        inner_params = declared(params, include_missing: true)
```

Comment:
I realize we do this elsewhere, but it's bad practice - maybe fix this and other instances?

This works, but is hacky and abuses the fact that the inner function still has access to outer variables. Below you check that the GET call succeeded with a 200, but then reach in for the value of `inner_params`. Be consistent, return `declared(...)` and examine the JSON response instead of relying on this.